### PR TITLE
CommitStoreV1.2 validation fix and RMN comments

### PR DESCRIPTION
**Pull request by matYang**
## Motivation
1. Allow CommitStoreV1.2 to accept 0 `DAGasPriceDeviationPPB`.
2. Make it clear in code on what changes impact RMN to avoid surprises.

### CODE REVIEW

**Review by gtklocker — COMMENTED** [review]
Would it be a good idea to add a notice to every direct/indirect dependency of structs/events you've already added the notice to?

File: core/services/ocr2/plugins/ccip/internal/ccipdata/commit_store_v1_2_0.go [hunk 1]
```diff
@@ -196,6 +193,7 @@ func (c CommitOffchainConfigV1_2_0) Validate() error {
 	if c.InflightCacheExpiry.Duration() == 0 {
 		return errors.New("must set InflightCacheExpiry")
 	}
+	// DAGasPriceDeviationPPB is not validated because it can be 0 on non-rollups
```
Comment (kaleofduty) [inline core/services/ocr2/plugins/ccip/internal/ccipdata/commit_store_v1_2_0.go:196]: How come?

Comment (matYang) [reply core/services/ocr2/plugins/ccip/internal/ccipdata/commit_store_v1_2_0.go:196]: on a non-rollup chain, e.g. Avax, the `DAGasPriceDeviationPPB` field is essentially meaningless right, in this case we can allow it to be unconfigured and default to `0`

**Comment by matYang** [comment]
> Would it be a good idea to add a notice to every direct/indirect dependency of structs/events you've already added the notice to?

Added comments to dependencies, with the exception of `Client.EVMTokenAmount`. Thought it may look weird for other developers to see an RMN comment in Client.sol. And since it's in client library, that struct will only be changed after significant discussions, likely involving research, an extra comment is probably unnecessary.

**Review by connorwstein — APPROVED** [review]

**Review by gtklocker — APPROVED** [review]

**Pull request merged**